### DISCUSSION TITLE
28.1.1

### DISCUSSION THREAD

**Discussion by njr0** [google/comprehensive-rust#1279] [General]
Even though I see it's straight out of the documentation, I think this could be made clearer.

The code is a bit odd because the first `result` is nothing (`()`).

This seems clearer to me:
```rust
use std::panic;

fn main() {
    let result = panic::catch_unwind(|| {
        "No problem here!"
    });
    assert!(result.is_ok());
    println!("{}", result.unwrap());

    let result = panic::catch_unwind(|| {
        panic!("oh no!");
    });
    assert!(result.is_err());
}
```
(Alternatively, could do an assertion on the unwrapped value of result, but printing it seems clearer to me.)

**Comment by djmitche** [comment]
-> #1352.
